tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/SDKBuildPropertiesResourceListener.java

```diff
@@ -0,0 +1,304 @@
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IResourceChangeEvent;
+import org.eclipse.core.resources.IResourceChangeListener;
+import org.eclipse.core.resources.IResourceDelta;
+import org.eclipse.core.resources.IResourceDeltaVisitor;
+import org.eclipse.core.resources.WorkspaceJob;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.jdt.core.ClasspathContainerInitializer;
+import org.eclipse.jdt.core.IClasspathContainer;
+import org.eclipse.jdt.core.IClasspathEntry;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.wst.common.project.facet.core.IFacetedProject;
+import org.eclipse.wst.common.project.facet.core.IProjectFacet;
+import org.eclipse.wst.common.project.facet.core.IProjectFacetVersion;
+import org.eclipse.wst.common.project.facet.core.ProjectFacetsManager;
+
+/**
+ * @author Simon Jiang
+ */
+public class SDKBuildPropertiesResourceListener implements IResourceChangeListener, IResourceDeltaVisitor
+{
+
+    private static final String ID_SDK_PROPERTIES_INVALID = "sdk-properties-invalid";
+
+
+    private final static Pattern buildPropertiesPattern = Pattern.compile("build.[\\w|\\W.]*properties");
+
+    public static boolean isLiferayProject( IProject project )
+    {
+        boolean retval = false;
+
+        try
+        {
+            IFacetedProject facetedProject = ProjectFacetsManager.create( project );
+
+            if( facetedProject != null )
+            {
+                for( IProjectFacetVersion facet : facetedProject.getProjectFacets() )
+                {
+                    IProjectFacet projectFacet = facet.getProjectFacet();
+
+                    if( projectFacet.getId().startsWith( "liferay." ) ) //$NON-NLS-1$
+                    {
+                        retval = true;
+
+                        break;
+                    }
+                }
+            }
+        }
+        catch( CoreException e )
+        {
+        }
+
+        return retval;
+    }
+
+    private void clearMarkers( IProject proj )
+    {
+        try
+        {
+            if( proj.isOpen() )
+            {
+                IMarker[] markers = proj.findMarkers( IMarker.PROBLEM, true, IResource.DEPTH_INFINITE );
+
+                for( IMarker marker : markers )
+                {
+                    if( marker.getAttribute( IMarker.SOURCE_ID ).equals( ID_SDK_PROPERTIES_INVALID ) )
+                    {
+                        marker.delete();
+                    }
+                }
+            }
+        }
+        catch( CoreException e )
+        {
+            ProjectCore.logError( e );
+        }
+    }
+
+    protected void processPropertiesFile( IFile buildPropertiesFile ) throws CoreException
```

Comment:
Need to make sure that the workspace SDK proejct is the same one that this file comes from.